ui/src/app/pages/workspace/workspace-edit.tsx

```diff
@@ -619,6 +669,31 @@ export const WorkspaceEdit = fp.flow(withRouteConfigData(), withCurrentWorkspace
       return researchPurpose.researchOutcomeList && researchPurpose.researchOutcomeList.length !== 0 ;
     }
 
+
+    updateResearchPurposeSummary(purposeSummary, value) {
+      if (this.state.researchSummaryProps[purposeSummary].warningMsg && value.length >= 50 ) {
+        this.setState(fp.set(['researchSummaryProps', purposeSummary, 'warningMsg'], false));
+      }
+      if (value.length > 1000) {
+        value = value.substring(0, 1000);
```

Comment:
Don't delete text the user has entered. Just prevent them from submitting it until it's a short enough text.

Reply:
I feel its better to restrict at 1000 rather than showing -ve and have additional checks ( i can remove the 1000 check at submit since it will be redundant)